File: problems/006/p006.go
package main

import "fmt"

func main() {

	var problem_result int = resolve(100)

	fmt.Printf("Result=%d\n", problem_result)

}

func resolve(factor int) int {

	var sum_square int
	var sum int

	for i := 1; i <= factor; i++ {
		sum_square += i * i
		sum += i
	}

	square_sum := sum * sum

	result := square_sum - sum_square

	return result
}

File: problems/006/p006.c

#include <stdio.h>

int resolve(int factor)
{
	int sumSquare = 1;
	int sum = 1;

	for (int i = 2; i <= factor; i++) {
		sumSquare += i * i;
		sum += i;
	}

	int squareSum = sum * sum;

	return squareSum - sumSquare;
}

int main()
{

	int result = resolve(100);

	printf("Result=%d\n", result);

	return 0;
}